From 990f7d546b618f29f790b0997c47fdb813e6ed09 Mon Sep 17 00:00:00 2001
From: Jonathan Pui <jpui@netlogicmicro.com>
Date: Mon, 2 May 2011 18:27:48 -0700
Subject: [PATCH 242/565] Order of node_vc#_mask parsing

Order of node_vc#_mask parsing is now consistent with other masks.

Based on Broadcom SDK 2.3.

Signed-off-by: Jonathan Pui <jpui@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/xlp/setup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/netlogic/xlp/setup.c b/arch/mips/netlogic/xlp/setup.c
index 22a6700..89e390f 100644
--- a/arch/mips/netlogic/xlp/setup.c
+++ b/arch/mips/netlogic/xlp/setup.c
@@ -452,7 +452,7 @@ static int fdt_process(void)
 	char domstr[32] = "";
 	uint32_t  i, na, ns, regs[16], entries, cpu_cells;
  	int index = 0;
-	uint32_t node_vc_mask[4] = {0};
+	uint32_t node_vc_mask[NLM_MAX_CPU_NODE] = {0};
 	extern uint32_t nlm_cpu_vc_mask[NLM_MAX_CPU_NODE*NLM_MAX_CPU_PER_NODE];
 	unsigned char buf[30];	
 	int j, id=0, k, tmp;
@@ -665,7 +665,7 @@ noloadermask:
 				}
 			}
 			else {	/* Get vc mask from the fdt */
-				for (j = 0; j < 4; j++) {
+				for (j = 3; j >= 0; j--) {
 					tmp = fdt32_to_cpu(node_vc_mask[j]);
 					for (k = 0; k < 8; k++) {
 						nlm_cpu_vc_mask[id++] = (tmp >> (k * 4)) & 0xf;
-- 
1.8.4.93.g57e4c17

